File: frontend/lib/models/channel_watch_state.dart
/// 채널별 "확인 상태" — 사용자가 마지막으로 확인한 최신 영상 시각을 추적한다.
///
/// 스탠다드(비로그인)는 이 상태를 단말 로컬(SharedPreferences)에만 저장한다.
/// 프리미엄(로그인) 확장 시, 이 모델 구조를 그대로 백엔드 테이블
/// (예: channel_watch_state)로 업로드/다운로드만 추가하면 동기화가 된다.
/// → 그래서 필드를 서버 컬럼과 1:1 매핑 가능하도록 단순/명시적으로 설계했다.
class ChannelWatchState {
  /// 채널 식별자 (YoutubeChannel.channelId)
  final String channelId;

  /// 마지막으로 "확인함" 처리한 시점의 최신 영상 게시일.
  /// 채널의 최신 영상 게시일이 이 값보다 나중이면 "새 영상 있음"으로 판단한다.
  final DateTime? lastSeenAt;

  /// 마지막으로 확인한 최신 영상의 videoId (보조 식별/디버깅용)
  final String? lastSeenVideoId;

  /// 이 상태가 갱신된 시각 (서버 동기화 시 충돌 해결 기준)
  final DateTime updatedAt;

  const ChannelWatchState({
    required this.channelId,
    this.lastSeenAt,
    this.lastSeenVideoId,
    required this.updatedAt,
  });

  ChannelWatchState copyWith({
    DateTime? lastSeenAt,
    String? lastSeenVideoId,
    DateTime? updatedAt,
  }) =>
      ChannelWatchState(
        channelId: channelId,
        lastSeenAt: lastSeenAt ?? this.lastSeenAt,
        lastSeenVideoId: lastSeenVideoId ?? this.lastSeenVideoId,
        updatedAt: updatedAt ?? this.updatedAt,
      );

  Map<String, dynamic> toJson() => {
        'channelId': channelId,
        'lastSeenAt': lastSeenAt?.toIso8601String(),
        'lastSeenVideoId': lastSeenVideoId,
        'updatedAt': updatedAt.toIso8601String(),
      };

  factory ChannelWatchState.fromJson(Map<String, dynamic> json) => ChannelWatchState(
        channelId: json['channelId'] as String? ?? '',
        lastSeenAt: json['lastSeenAt'] != null ? DateTime.tryParse(json['lastSeenAt'] as String) : null,
        lastSeenVideoId: json['lastSeenVideoId'] as String?,
        updatedAt: json['updatedAt'] != null
            ? (DateTime.tryParse(json['updatedAt'] as String) ?? DateTime.fromMillisecondsSinceEpoch(0))
            : DateTime.fromMillisecondsSinceEpoch(0),
      );
}

File: backend/src/main/java/com/litten/note/youtube/ChannelWatchState.java
package com.litten.note.youtube;

import com.litten.common.dynamic.BaseEntity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import jakarta.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 채널별 "확인 상태" 엔티티 (note_channel_watch_state).
 *
 * 프론트 ChannelWatchState 모델과 1:1 매핑.
 * 사용자가 마지막으로 확인한 최신 영상 시각을 추적해 "새 영상(new)" 여부를 판단한다.
 *   - 채널 최신 영상 게시일 > last_seen_at → 새 영상 있음
 *
 * 프리미엄(로그인) 동기화 대상. (member_id, channel_id) 단위로 유일.
 * 비로그인은 프론트 로컬(SharedPreferences)에만 저장하고 서버에 올리지 않는다.
 */
@Getter
@Setter
@NoArgsConstructor
@Entity(name = "ChannelWatchState")
@Table(name = "note_channel_watch_state",
    uniqueConstraints = @UniqueConstraint(
        name = "unique_watch_member_channel",
        columnNames = {"member_id", "channel_id"}
    ),
    indexes = {
        @Index(name = "idx_watch_member", columnList = "member_id")
    }
)
public class ChannelWatchState extends BaseEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", columnDefinition = "BIGINT NOT NULL AUTO_INCREMENT COMMENT 'PK'")
    private Long id;

    @Column(name = "member_id", nullable = false,
            columnDefinition = "VARCHAR(128) NOT NULL COMMENT '회원 ID (로그인 사용자)'")
    private String memberId;

    @Column(name = "channel_id", nullable = false,
            columnDefinition = "VARCHAR(64) NOT NULL COMMENT '유튜브 채널 ID'")
    private String channelId;

    @Column(name = "last_seen_at",
            columnDefinition = "TIMESTAMP NULL COMMENT '마지막 확인 시점의 최신 영상 게시일'")
    private LocalDateTime lastSeenAt;

    @Column(name = "last_seen_video_id",
            columnDefinition = "VARCHAR(50) NULL COMMENT '마지막 확인한 최신 영상 videoId (보조 식별)'")
    private String lastSeenVideoId;

    @Column(name = "synced_at", nullable = false,
            columnDefinition = "TIMESTAMP NOT NULL COMMENT '상태 갱신 시각 (동기화 충돌 해결 기준 — 프론트 updatedAt)'")
    private LocalDateTime syncedAt;
}
